data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -154,13 +166,60 @@ private long calculateLongestThreadWaitingTime() {
                 orElse(1L);
     }
 
-    private Properties getConsumerProperties(TopicConfig topicConfig) {
+    public String getBootStrapServersForMsk(final AwsIamAuthConfig awsIamAuthConfig, final AwsConfig awsConfig) {
+        ClientConfiguration clientConfiguration = new ClientConfiguration();
+        AWSCredentialsProvider credentialProvider;
+        if (awsIamAuthConfig == AwsIamAuthConfig.DEFAULT) {
+            credentialProvider = new DefaultAWSCredentialsProviderChain();
+        } else if (awsIamAuthConfig == AwsIamAuthConfig.ROLE) {
+            String sessionName = "data-prepper-kafka-session"+UUID.randomUUID();
+            credentialProvider = new STSAssumeRoleSessionCredentialsProvider.Builder(awsConfig.getStsRoleArn(), sessionName).build();
```

Comment:
Do we need to wrap it in try/catch and log the exception? Also if it fails will we retry with backoff? like we retry from opensearch sink.

Reply:
This is just a builder. I don't think it can throw exception. The `kafkaClient.getBootstrapBrokers` below may result in exceptions and some of which are retry-able. I will add that logic.